libraries/standard/mqtt/include/mqtt_lightweight.h

```diff
@@ -400,57 +433,69 @@ MQTTStatus_t MQTT_GetPublishPacketSize( const MQTTPublishInfo_t * pPublishInfo,
  * consider using #MQTT_SerializePublishHeader, which will serialize
  * only the PUBLISH header into the buffer.
  *
+ * #MQTT_GetPublishPacketSize should be called with @p pPublishInfo before
+ * invoking this function. The @p remainingLength was calculated from the
+ * parameters in @p pPublishInfo using function #MQTT_GetPublishPacketSize.
+ *  @p pPublishInfo and @p remainingLength are valid only if
+ * #MQTT_GetPublishPacketSize returned #MQTTSuccess.
+ *
  * @param[in] pPublishInfo MQTT PUBLISH packet parameters.
  * @param[in] packetId packet ID generated by #MQTT_GetPacketId.
- * @param[in] remainingLength Remaining Length provided by #MQTT_GetConnectPacketSize.
- * @param[out] pBuffer Buffer for packet serialization.
+ * @param[in] remainingLength Remaining Length provided by #MQTT_GetPublishPacketSize.
+ * @param[out] pFixedBuffer Buffer for packet serialization.
  *
- * @return #MQTTNoMemory if pBuffer is too small to hold the MQTT packet;
+ * @return #MQTTNoMemory if pFixedBuffer is too small to hold the MQTT packet;
  * #MQTTBadParameter if invalid parameters are passed;
  * #MQTTSuccess otherwise.
  */
 MQTTStatus_t MQTT_SerializePublish( const MQTTPublishInfo_t * pPublishInfo,
                                     uint16_t packetId,
                                     size_t remainingLength,
-                                    const MQTTFixedBuffer_t * pBuffer );
+                                    const MQTTFixedBuffer_t * pFixedBuffer );
 
 /**
  * @brief Serialize an MQTT PUBLISH packet header in the given buffer.
  *
- * This function serializes PUBLISH header in to the given buffer. Payload
+ * This function serializes PUBLISH header in to the given buffer. The payload
  * for PUBLISH will not be copied over to the buffer. This will help reduce
- * the memory needed for the buffer and avoid an unwanted copy operataion of
- * PUBLISH payload into the buffer. If payload also would need to be part of
+ * the memory needed for the buffer and avoid an unwanted copy operataion of the
```

Comment:
spelling: _operation_

Also, is this directory/file being spell-checked?  It should be.

Reply:
It is part of this sprint to spell check all libraries at once